### ISSUE TITLE
Do not override values which are defined as Variable replacement Expressions.

### ISSUE THREAD

**Issue by zebkailash** [auth0/auth0-deploy-cli#292]
### Describe the problem you'd like to have solved

We are trying to automate the Auth0 configuration deployment using Github Actions. We have created a workflow with branches development, staging, production. Changes to dev tenant will be synced to local repository manually by developer. We have made use for variable substitution in the clients wherever required like below
```
{
    "AUTH0_TENANT_NAME": "dev-tenantname",
    "MANAGEMENT_API_AUDIENCE": "https://dev-tenantname.eu.auth0.com/api/v2/",
    "API_IDENTIFIER": "https://api.example.io",
    "APP_CALLBACKS": [
      "http://localhost:3000",
      "https://dev-example.com/",
    ]
  }
```
And used with `##APP_CALLBACKS##` in the client.json wherever those values are required.

## Problem:
How to sync further changes once one deployment cycle is finished and we want to incorporate more changes made to dev env to the local repo? Right now it overrides the configuration file with whatever values are configured in dev tenant and all the previous changes are lost and then it's a manual process to make those changes again which defeats the whole motive of the automation process.

### Describe the ideal solution
If a client where variable substitution expression is used it should merge with existing change (change with ##vars##) over the new changes from dev tenant that will avoid any wrong deployment configuration being deployed to other environment.

### Ideal Solution
Merge the new config over the existing config to avoid replacing the substituttion vars. 

## Alternatives and current work-arounds
Some kind of pre push scripts to match before pushing to check if values defined in config has matched exactly in client config.

### Additional context
"auth0-deploy-cli": "^5.3.1",

**Comment by willvedd** [comment]
Thanks for opening this issue, I understand that this can be quite a frustrating problem. However, similarly to #125, I'm closing this in favor of #328 which has a more substantial conversation.
